app/controllers/contracts_controller.rb

```diff
@@ -1,8 +1,10 @@
 class ContractsController < ApplicationController
-  before_action :set_contract, only: [:show, :reports]
+  before_action :set_contract, only: [:show, :reports, :update]
   def index
     @contracts = Contract.joins(:project).includes(:full_reports, :project)
       .order('projects.name ASC, contracts.name ASC')
+    @states = Contract.aasm.states
+    @contracts = @contracts.with_status(params[:state]) if params[:state].present?
```

Comment:
I'd default to live contracts, this way the list is automatically more manageable!

Reply:
Yes, but there is a small nuance: default is not stored in the DB for the records already created. I was thinking about using "proposal" by default, for this reason, to make sure that the state will change and will be stored into the DB. Also, if you create a new contract, it will be live automatically if it's the default. What do you think?

Reply:
On the setup is OK, what I meant is set the default filter to `live`. Sorry I wasn't very clear!
But on the whole the flow makes sense, and we will want to add projects as proposal. I would just set the filter to have live as the default.